pf4j/src/main/java/org/pf4j/Extension.java

```diff
@@ -47,4 +47,16 @@
      */
     Class<? extends ExtensionPoint>[] points() default {};
 
+    /**
+     * An array of plugin IDs, that have to be available in order to load this extension.
+     * The {@link AbstractExtensionFinder} won't load this extension, if these plugins are not
+     * available / started at runtime.
+     * <p>
+     * Notice: This feature requires the optional <a href="https://asm.ow2.io/">ASM library</a>
+     * to be available on the applications classpath and has to be explicitly enabled via
+     * {@link AbstractExtensionFinder#setCheckForExtensionDependencies(boolean)}.
+     *
+     * @return plugin IDs, that have to be available in order to load this extension
+     */
+    String[] plugins() default {};
```

Comment:
Is it not more clear to rename `plugins` attribute to `requireOptionalPlugins`? It's a little bit long as name but it reflects the semantic.

Reply:
My initial idea was, to call the `points` attribute `provides` (introduced with pull request #265) and to call the `plugins` attribute `requires`. According to my personal taste

```java
@Extension(
  provides = ContactsFormExtensionPoint.class, 
  requires = {CalendarPlugin.ID, ContactsPlugin.ID}
) 
```

looks most natural to me. It's kind of short and it's more or less self-explanatory. If a developer starts using PF4J, he would have to read the documentation anyway. I guess a longer (but more precise) attribute name wouldn't really change that.

What do you think about this naming?

> Is it not more clear to rename `plugins` attribute to `requireOptionalPlugins`? It's a little bit long as name but it reflects the semantic.

In general this attribute can also be used for non-optional plugins. Maybe in the future it might make sense to extend this feature for a particular version check. An extension might depend on another / later plugin version compared to the plugin version defined via `MANIFEST.MF` / `plugin.properties`.

In case of future improvements on this feature, a name like `requireOptionalPlugins` might be misleading or had to be changed afterwards (which would break backwards compatibility). Therefore I would prefer a more general name like `plugins` or `requires`.

---

But I don't insist on a particular name for this attribute. I'm fine, if you prefer `requireOptionalPlugins` over `plugins` or `requires`. Let me know, which naming you prefer and I'm going to update the pull request accordingly.

Reply:
I preserved `plugins` name because is small and maybe in the future we will inject more business around this attribute.

> In general this attribute can also be used for non-optional plugins.

Here is not clear for me. See my above [example](https://github.com/pf4j/pf4j/pull/270#discussion_r247104002), I don't think that is possible to declare in `E1` extension from plugin `P1` that it requires a plugin `P3` that is not declared as dependency (optional or not) in `P1`.

Reply:
> > In general this attribute can also be used for non-optional plugins.
> 
> Here is not clear for me. See my above [example](https://github.com/pf4j/pf4j/pull/270#discussion_r247104002), I don't think that is possible to declare in `E1` extension from plugin `P1` that it requires a plugin `P3` that is not declared as dependency (optional or not) in `P1`.

That's true. I meant something different. `P1` might have a non-optional dependency on `P2` with version 1.0. But an extension provided by `P1` might depend on `P2` with version 1.1, because the corresponding extension point was added later. If `P2` is only available in version 1.0, the extension is not loaded but `P1` is still usable.

I haven't fully thought through this idea. I also don't need this at the moment. It was only meant as an example for a possible future improvement on this feature.